library_methods.c

```diff
@@ -35,19 +35,18 @@ struct song_node * search_artist(struct song_node * library, char * artist) {
 void print_letter(struct song_node ** library, char letter){
     int index = letter - 'a' + 1;
     if (index < 0) {index = 0;}
-    printf("%d\n", index);
     printf("%c list\n", letter);
     print_list(library[index]);
 }
 
 // matthew
 // Print out all the songs of a certain artist
-void print_artist(struct song_node * library, char * artist) {
+void print_artist(struct song_node ** library, char * artist) {
     int index = artist[0] - 'a' + 1;
     if (index < 0) {index = 0;}
     
     printf("Printing [%s]\n", artist);
-    struct song_node *foundNode = find_artist(&library[index], artist);
+    struct song_node *foundNode = find_artist(library[index], artist);
     for(; foundNode->artist!=artist; foundNode=foundNode->next) {
         print_node(foundNode);
```

Comment:
I think we can obfuscate the need to move print statements in find_artist by not finding artist in this case. rather we could just start from the first node in library[index].
NTTSTT in exchange for a slightly less efficient algorithm, or slightly more efficient algorithm in exchange for disappointing holmes?